File: typescript/src/address/search/searchResult.ts
import { RestApiAddressSearchResponse } from "../../server/address/search/restApiAddressSearchResponse";
import { AddressConfidence, lookupConfidence } from "../addressConfidence";
import { restApiResponseToSuggestion, SearchSuggestion } from "./searchSuggestion";

export type SearchResult = {
    moreResultsAvailable: boolean;
    confidence: AddressConfidence;
    suggestionsKey: string;
    suggestionsPrompt: string;
    suggestions: SearchSuggestion[];
    referenceId?: string;
}

export function restApiResponseToSearchResult(response: RestApiAddressSearchResponse): SearchResult {
        
    const apiResult = response.result;
    if (apiResult) {
        return {
            moreResultsAvailable: apiResult.more_results_available ?? false,
            confidence: lookupConfidence(apiResult.confidence),
            suggestionsKey: apiResult.suggestions_key??"",
            suggestionsPrompt: apiResult.suggestions_prompt??"",
            suggestions: apiResult.suggestions?.map(suggestion => restApiResponseToSuggestion(suggestion)) ?? [],
            referenceId: response.referenceId,
        }
    } else {
        return {
            moreResultsAvailable: false,
            confidence: AddressConfidence.Unknown,
            suggestionsKey: "",
            suggestionsPrompt: "",
            suggestions: [],
            referenceId: response.referenceId
        }
    }
}

File: csharp/DVSClient/DVSClient/address/search/SearchResult.cs
using DVSClient.Common;
using DVSClient.Server.Address.Search;

namespace DVSClient.Address.Search
{
    public class SearchResult
    {
        public bool MoreResultsAvailable { get; }
        public AddressConfidence? Confidence { get; }
        public string SuggestionsKey { get; }
        public string SuggestionsPrompt { get; }
        public IEnumerable<SearchSuggestion> Suggestions { get; } = new List<SearchSuggestion>();

        public SearchResult(RestApiAddressSearchResponse response)
        {
            var apiResult = response.Result;
            if (apiResult != null)
            {
                MoreResultsAvailable = apiResult.IsMoreResultsAvailable() ?? false;
                Confidence = apiResult.Confidence?.GetEnumValueFromJsonName<AddressConfidence>();
                SuggestionsKey = apiResult.SuggestionsKey ?? string.Empty;
                SuggestionsPrompt = apiResult.SuggestionsPrompt ?? string.Empty;
                Suggestions = apiResult.Suggestions?.Select(s => new SearchSuggestion(s)).ToList() ?? new List<SearchSuggestion>();
            }
            else
            {
                MoreResultsAvailable = false;
                Confidence = null;
                SuggestionsKey = string.Empty;
                SuggestionsPrompt = string.Empty;
                Suggestions = new List<SearchSuggestion>();
            }
        }
    }
}